staging/src/k8s.io/api/core/v1/types.go

```diff
@@ -3085,6 +3093,15 @@ type PodStatus struct {
 	QOSClass PodQOSClass `json:"qosClass,omitempty" protobuf:"bytes,9,rep,name=qosClass"`
 }
 
+// PodIPInfo represents IP information of a pod.
+type PodIPInfo struct {
+	// IP address allocated to the pod. Routable at least within the cluster.
+	IP string `json:"ip,omitempty" protobuf:"bytes,1,opt,name=ip"`
+	// Arbitrary metadata associated to the allocated IP.
```

Comment:
We need more spec on this - what format do keys and values take, how long can each key and value be?  How many can there be?  What is the limit on the total size of this map.

We should also think about whether we want first-class "flags" on IPs, such as "probe me".  Is that something that Kubelet can know, or plugins can produce?  Maybe probe IPs are always the [0] ?

How does a field end up in here?  It is assumed to be 3rd-party things?